### PR TITLE
Apply ListTile colors to leading and trailing text widgets

### LINKED ISSUE #73781: Disabled ListTile theme colors are not applied to the leading and trailing text widgets
## Steps to Reproduce

From documentation for `enabled` property we have
>Whether this list tile is interactive.
> If false, this list tile is styled with the disabled color from the current Theme and the onTap and onLongPress callbacks are inoperative.

The disabled color is no applied to **leading** and **trailing**
```dart
ListTile(
      leading: FlutterLogo(size: 56.0),
      title: Text('Two-line ListTile'),
      subtitle: Text('Here is a second line'),
      trailing: Text('Trailing'),
      enabled: false,
    );
```

**Expected results:** 

The disabled theme must be applied to **leading** and **trailing** too

**Actual results:** 

image

### PR DESCRIPTION
**Pull request by shihaohong**
Applies text colors to Text widgets present in leading and trailing widget subtrees. 

I opted for `theme.textTheme.bodyText2` as the basis for the text styles since that would apply the default material text styles (such as size and font) other than the color behaviors, which would be the least breaking route of applying the disabled/selected/default colors to the subtrees.

Fixes https://github.com/flutter/flutter/issues/73781.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/list_tile_test.dart [hunk 1]
```diff
@@ -2083,4 +2083,113 @@ void main() {
...
+
+    await tester.pumpWidget(buildFrame());
+    // Enabled color should be default bodyText2 color.
+    expect(textColor(leadingKey), theme.textTheme.bodyText2!.color);
+    expect(textColor(trailingKey), theme.textTheme.bodyText2!.color);
+
+    await tester.pumpWidget(buildFrame(selected: true));
+    // Wait for text color to animate.
+    await tester.pumpAndSettle();
+    // Selected color should be default bodyText2 color.
```
Comment (xu-baolin) [inline packages/flutter/test/material/list_tile_test.dart:2126]: primaryColor？

File: packages/flutter/lib/src/material/list_tile.dart [hunk 2]
```diff
@@ -1076,6 +1076,12 @@ class ListTile extends StatelessWidget {
       : style.copyWith(color: color);
   }
 
+  TextStyle _trailingAndLeadingTextStyle(ThemeData theme, ListTileTheme? tileTheme) {
+    final TextStyle style = theme.textTheme.bodyText2!;
+    final Color? color = _textColor(theme, tileTheme, theme.textTheme.bodyText2!.color);
```
Comment (xu-baolin) [inline packages/flutter/lib/src/material/list_tile.dart:1081]: nit: 
```dart
final Color? color = _textColor(theme, tileTheme, style.color);
```

**Review by xu-baolin — APPROVED** [review]
LGTM

**Comment by xu-baolin** [comment]
Another rare scene, if the `ListTile.title` or `ListTile.subtitle` is an icon widget, what should we do?

**Comment by shihaohong** [comment]
@xu-baolin Hmm, I think we definitely want to address that case as well at some point. I'm surprised that that hasn't come up as a problem yet. I'll file an issue about it (https://github.com/flutter/flutter/issues/73933)

**Pull request merged**